pip/pip-368.md

```diff
@@ -0,0 +1,148 @@
+# PIP-368: Support client properties for the lookup
+
+# Background knowledge
+
+## How Pulsar Lookup Works
+
+Before producing or consuming messages, a Pulsar client must first find the broker responsible for the topic. This
+happens through the lookup service. The client sends a `CommandLookupTopic` request with the topic name to the lookup
+service, which then returns the responsible broker. The client then interacts with this broker to produce or consume
+messages.
+
+On the broker side, users can customize lookup logic by setting a custom load manager in the `loadManagerClassName`
+configuration.
+
+# Motivation
+
+Currently, the lookup process uses only the topic name as its parameter. However, to enhance this process, it's
+beneficial for clients to provide additional information. This could be done by introducing the `properties` field in
+the client configuration. The client can then send these properties to the broker during the lookup process.
+
+Here is a scenario for using the client properties for the lookup:
+Assuming there are two brokers that broker-0 configures the "rack" property with "A" and broker-1 configures the "rack"
+property with "B". By using the lookup properties, clients can supply rack information during the lookup, enabling the
```

Comment:
Even if a broker configures the "rack" property, there is no way to let a broker know another broker's "rack" property.

Reply:
This part is not very familar to me, is the broker's properties not stored in the metadata?

Reply:
Thanks. I added the support of the `lookupProperties` on the broker side. The broker lookup properties will be saved in the metadata store. So the broker can now access the lookup properties for all other brokers. This will help implement this rack-aware lookup scenario.

> is the broker's properties not stored in the metadata?

No. So I updated this proposal to make the broker properties that start with `lookupPropertyPrefix` be able to be stored in the metadata store.